src/read.c

```diff
@@ -3707,25 +3695,31 @@ static avifResult avifDecoderDataFindAlphaItem(avifDecoderData * data,
         }
     }
     assert(alphaItemCount == colorItemCount);
-    // Figure out the last used itemID.
-    avifResult result = AVIF_RESULT_OK;
-    const uint32_t lastID = colorItem->meta->items.item[colorItem->meta->items.count - 1]->id;
-    if (lastID == UINT32_MAX) {
-        // In the improbable case where the last ID is the maximum one, ids cannot be kept ordered.
-        avifDiagnosticsPrintf(diag,
-                              "Cannot set an itemID for alpha that fits the increasing "
-                              "order as the maximum possible ID is in use.");
+    // Find an unused ID.
+    avifResult result;
+    if (colorItem->meta->items.count >= UINT32_MAX - 1) {
+        // In the improbable case where all IDs are used.
         result = AVIF_RESULT_DECODE_ALPHA_FAILED;
     } else {
-        *alphaItem = avifMetaFindItem(colorItem->meta, lastID + 1, diag); // Create new empty item.
-        if (*alphaItem == NULL) {
-            result = AVIF_RESULT_OUT_OF_MEMORY;
-        }
+        uint32_t newItemID = 0;
+        avifBool isUsed;
+        do {
+            ++newItemID;
+            isUsed = AVIF_FALSE;
+            for (uint32_t i = 0; i < colorItem->meta->items.count; ++i) {
+                if (colorItem->meta->items.item[i]->id == newItemID) {
+                    isUsed = AVIF_TRUE;
+                    break;
+                }
+            }
+        } while (isUsed && newItemID != 0);
+        *alphaItem = avifMetaFindItem(colorItem->meta, newItemID); // Create new empty item.
```

Comment:
This is fine because

https://github.com/AOMediaCodec/libavif/blob/e6edfd7f04ca7fd9305401d04a7a853208cc73c4/src/read.c#L773-L777

Reply:
Yes. If `newItemID` is 0, `avifMetaFindItem(colorItem->meta, newItemID)` fails and returns `NULL`. So we will set `result` to `AVIF_RESULT_OUT_OF_MEMORY` in the next line. `AVIF_RESULT_OUT_OF_MEMORY` is not the right error code for this condition, but at least we will fail rather than keep going.

However, if all the item IDs in the `colorItem->meta->items` array are nonzero and distinct (I believe this is an invariant that we maintain), then `newItemID` cannot possibly be zero. So we could add `assert(newItemID != 0)` after the do-while loop to indicate our belief.

Note: I suggested removing `&& newItemID != 0` from the condition of the do-while loop, but Vincent was uncomfortable with a potential infinite loop if we fail to maintain the invariant.

Reply:
I just realized that my previous comment is wrong:

> However, if all the item IDs in the `colorItem->meta->items` array are nonzero and distinct (I believe this is an invariant that we maintain), then `newItemID` cannot possibly be zero. [...]

All we need to prove `newItemID` cannot be zero is that `colorItem->meta->items.count < UINT32_MAX - 1`. If there are fewer than `UINT32_MAX - 1` elements in the `colorItem->meta->items` array, then there must be a nonzero `uint32_t` value that is not an item ID in the `colorItem->meta->items` array.

Since the `colorItem->meta->items.count < UINT32_MAX - 1` condition is guaranteed by the check at line 3700, the only way the do-while loop may loop infinitely is that the loop itself has a bug. The do-while loop is short and can be reviewed thoroughly. So I think the `&& newItemID != 0` check can be safely omitted.